backend/addcorpus/models.py

```diff
@@ -227,6 +229,12 @@ class CorpusConfiguration(models.Model):
         help_text='name of the field that specifies the language of documents (if any);'
             'required to use "dynamic" language on fields',
     )
+    source_data = models.JSONField(
```

Comment:
Why use a `JSONField` here and for `extract_options`. Would it not be desirable to limit and validate choices? E.g., that this could be `csv/json/xml/` etc.? From the name, I actually also expected `source_data` to define the location of the source data, not its format, so perhaps renaming to `source_data_format` would be clearer.

Reply:
I think I agree. If we add this data as flat fields, we only need `source_data_delimiter` on the CorpusConfiguration and `extract_options_column` on the Field model. Both would be `CharField`. (File type is included in the JSON for forwards compatibilty, but there is only one option.)

@JeltevanBoheemen what do you think?

Reply:
We chose a `JSONField` to be flexible for future extensions to both set of options, which is a very conceivable situation. 
The described validation is applied, through the JSON schema. If we support more source_data_formats, it would simply be a matter of adding them to the enum in the schema. 
I still stand behind this implementation, but agree that we should document it better.

Reply:
I agree that the JSON field already has fine validation, but what convinced me is that JSON fields are difficult to work with in the Django admin. If you encode the actual options, the delimiter can just be a dropdown, instead of nested in a JSON. I do think having a nice(-ish) admin form is a good intermediary goal before we build the frontend.

Also, if we do add more options, we can always migrate the existing data into a JSON field.

Reply:
Point taken on the admin interface